### PR TITLE
Test bin.str.validchars

### PR DESCRIPTION
**Pull request by kazarmy**
Added/fixed tests for radare/radare2#14503.

### CODE REVIEW

**Comment by radare** [comment]
.

**Comment by radare** [comment]
is this fixing the issue wihout passing  any extra option to r2? because that is the expected output imho

File: new/db/cmd/cmd_i [hunk 1]
```diff
@@ -3555,7 +3574,7 @@ NAME=iz/izz bin.str.purge
 FILE=../bins/elf/analysis/ls2
 EXPECT=<<EOF
 --1--
-1060 0x00012420 0x00412420   5   6 (.text) ascii AWAVA
+1063 0x00012420 0x00412420   5   6 (.text) ascii AWAVA
```
[the remark this answers is not in the document]
Comment (kazarmy) [reply new/db/cmd/cmd_i]: Different ordinal

File: new/db/cmd/cmd_i [hunk 2]
```diff
@@ -3410,7 +3410,7 @@ RUN
 NAME=iz utf16le
 FILE=../bins/elf/strenc
 EXPECT=<<EOF
-005 0x0000224a 0x0040224a  56 116 (.rodata) utf16le utf16le> \\u00a2\\u20ac\\U00010348 in green:\e[32m ¢€𐍈 \e[0m\n blocks=Basic Latin,Latin-1 Supplement,Currency Symbols,Gothic
+005 0x00002248 0x00402248  57 118 (.rodata) utf16le \nutf16le> \\u00a2\\u20ac\\U00010348 in green:\e[32m ¢€𐍈 \e[0m\n blocks=Basic Latin,Latin-1 Supplement,Currency Symbols,Gothic
```
[the remark this answers is not in the document]
Comment (kazarmy) [reply new/db/cmd/cmd_i]: Yes but probably a necessary evil

**Comment by kazarmy** [comment]
doing this automatically without a new config var will probably require significant work to `r_utf_block_list`, in particular to count the number of chars per block

**Comment by radare** [comment]
So its fine to merge now?

**Comment by kazarmy** [comment]
No unless the fully automated version can't be made to work.

**Comment by kazarmy** [comment]
A problem is that a byte in a wrong place misaligns all the wide chars after it, which can result in a slew of (nonexistent) CJK chars. There might be a workaround for this though.

**Comment by kazarmy** [comment]
> So its fine to merge now?

Don't merge yet.

**Comment by kazarmy** [comment]
Closed due to closure of radare/radare2#14503.

**Pull request closed without merging**